packages/ui/src/memberships/modals/BuyMembershipModal/BuyMembershipFormModal.tsx

```diff
@@ -177,19 +177,53 @@ export const BuyMembershipForm = ({
     }
   }, [data?.membershipsConnection.totalCount])
 
-  const isFormValid = !isUploading && form.formState.isValid && (!isValidator || validatorAccounts?.length)
+  const isFormValid =
+    !isUploading &&
+    form.formState.isValid &&
+    (!isValidator ||
+      (validatorAccountCandidates?.filter((account) => !account || !isValidValidatorAccount(account)).length === 0 &&
+        validatorAccountCandidates?.filter((account) => !!account).length))
   const isDisabled =
     type === 'onBoarding' && process.env.REACT_APP_CAPTCHA_SITE_KEY ? !captchaToken || !isFormValid : !isFormValid
 
   const addValidatorAccount = () => {
-    if (validatorAccountCandidate && isValidValidatorAccount) {
-      setValidatorAccounts([...new Set([...validatorAccounts, validatorAccountCandidate])])
-      form?.setValue('validatorAccountCandidate' as keyof MemberFormFields, undefined)
-    }
+    setValidatorAccounts([...(validatorAccountCandidates ?? []), undefined])
+    form?.setValue('validatorAccountCandidate' as keyof MemberFormFields, undefined)
   }
 
   const removeValidatorAccount = (index: number) => {
-    setValidatorAccounts([...validatorAccounts.slice(0, index), ...validatorAccounts.slice(index + 1)])
+    validatorAccountCandidates &&
+      setValidatorAccounts([
+        ...validatorAccountCandidates.slice(0, index),
+        ...validatorAccountCandidates.slice(index + 1),
+      ])
+  }
+
+  const setValidatorAccounts = (accounts: (Account | undefined)[]) => {
+    form?.setValue('validatorAccountCandidates' as keyof MemberFormFields, [])
+    accounts.map((account, index) => {
+      form?.register(('validatorAccountCandidates[' + index + ']') as keyof MemberFormFields)
+      form?.setValue(('validatorAccountCandidates[' + index + ']') as keyof MemberFormFields, account)
```

Comment:
I don't think React Hook Form should be used this way (registering fields again and again). Instead why not keeping `validatorAccounts` an internal state in the component ?

```ts
const [validatorAccounts, setValidatorAccounts] = useState<(Account | undefined)[]>([])
```